database-restore-postgres.sh

```
@@ -0,0 +1,289 @@
+    create_schema_sql="DROP SCHEMA IF EXISTS $schema CASCADE; CREATE SCHEMA $schema;"
+    
+    if [[ "$use_docker" == "true" ]]; then
+        echo "$create_schema_sql" | docker exec -i -e PGPASSWORD="$pass" "$service" "$psql_cmd" -U "$user" -h "$host" -p "$port" -d "$database"
+        docker exec -i -e PGPASSWORD="$pass" "$service" "$psql_cmd" -U "$user" -h "$host" -p "$port" -d "$database" < "$sql_file"
+    else
+        echo "$create_schema_sql" | PGPASSWORD="$pass" "$psql_cmd" -U "$user" -h "$host" -p "$port" -d "$database" 
+        PGPASSWORD="$pass" "$psql_cmd" -U "$user" -h "$host" -p "$port" -d "$database" -f "$sql_file"
+    fi
+    
+    # Check if restore was successful
+    if [[ $? -ne 0 ]]; then
```

Shellcheck (0.10.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)